script/cadence/contracts/FlowColdStorageProxy.cdc

```diff
@@ -141,14 +138,16 @@ pub contract FlowColdStorageProxy {
     //
     // And, finally, the FlowColdStorageProxy.Vault itself is made directly
     // accessible via /public/flowColdStorageProxyVault.
-    pub fun setup(payer: AuthAccount, publicKey: [UInt8]): Address {
+    access(all) fun setup(payer: AuthAccount, publicKey: [UInt8]): Address {
```

Comment:
tbh I really hate the idea of passing around account objects - pretty big anti-pattern as it exposes . But if we are going to simply refactor here this will need to be an entitled account where `payer` is:

```cadence
payer: auth(BorrowValue) &Account
```

If we have the option without disrupting anything upstream, it would be best to perform this setup in the context of a transaction. I guess the only consideration there would be exposing the created account to the transaction scope, making this call context dependent 🤔

Reply:
I just added an update which I think covers the first part of your feedback. For the change to the transaction I can see it's only used from [this transaction](https://github.com/onflow/rosetta/blob/main/script/cadence/transactions/create-proxy-account.cdc#L7) so i added the same there. However, I suspect that one of these two isn't right/needed but I couldn't find anything in the Cadence entitlements docs or otherwise in forum to clarify what I should be doing. Would appreciate your feedback